runtime/src/main/scala/akka/grpc/GrpcProtocol.scala

```diff
@@ -0,0 +1,112 @@
+package akka.grpc
+
+import akka.NotUsed
+import akka.grpc.GrpcProtocol.{ GrpcProtocolReader, GrpcProtocolWriter }
+import akka.grpc.internal.{ GrpcProtocolNative, GrpcProtocolWeb, GrpcWebTextProtocol }
+import akka.http.javadsl.{ model => jmodel }
+import akka.http.scaladsl.model.{ ContentType, HttpHeader }
+import akka.http.scaladsl.model.HttpEntity.ChunkStreamPart
+import akka.stream.scaladsl.Flow
+import akka.util.ByteString
+
+import scala.util.Try
+
+/**
+ * A variant of the gRPC protocol - e.g. gRPC and gRPC-Web
+ */
+trait GrpcProtocol {
+
+  /** The canonical media type to use for this protocol variant */
+  val contentType: ContentType.Binary
+
+  /** The set of media types that can identify this protocol variant (e.g. including an implicit +proto) */
+  val mediaTypes: Set[jmodel.MediaType]
+
+  /**
+   * Constructs a protocol writer for writing gRPC protocol frames for this variant
+   * @param codec the compression codec to encode data frame bodies with.
+   */
+  def newWriter(codec: Codec): GrpcProtocolWriter
+
+  /**
+   * Constructs a protocol reader for reading gRPC protocol frames for this variant.
+   * @param codec the compression codec to decode data frame bodies with.
+   */
+  def newReader(codec: Codec): GrpcProtocolReader
+}
+
+/**
+ * Core definitions for gRPC protocols.
+ */
+object GrpcProtocol {
+
+  private val grpcVariants: Seq[GrpcProtocol] = Seq(GrpcProtocolNative, GrpcProtocolWeb, GrpcWebTextProtocol)
```

Comment:
The protocol classes could have more consistent names (eg. `GrpcNativeProtocol`, `GrpcWebProtocol`)

Reply:
Yeah, apologies for that - IntelliJ kept breaking during rename refactors and I missed that.